src/vtarget-out/inv-syn/inv_syn.cc

```diff
@@ -0,0 +1,633 @@
+            // 
+            for (const auto & expr : exprs.exprs) {
+              ILA_ASSERT( expr.find("=") == expr.npos ) 
+                << "Bug: should be pure name."
+                << "but get: " << expr;
+
+              wn_asst_item.insert(expr);
+            }
+          }
+          else
+            ILA_ASSERT(false) << "Unknown dspt:" << dspt;
+
+
+        } // end of parsing asserts
+
+        for (auto && pbname_prob_pair : _problems.assumptions) {
+          const auto & dspt = pbname_prob_pair.first;
+          const auto & exprs = pbname_prob_pair.second;
+          for (const auto &expr : exprs.exprs) {
+            ILA_ASSERT( expr.find("=") == expr.npos ) // no "="
+              << "Bug: should be pure name."
+              << " but get: " << expr
+              << " dspt: " << dspt; 
+
+            if(dspt == "invariant_assume")
+              wn_amc_design_item.insert(expr);
+            else if(
+              dspt == "noreset" ||
+              dspt == "funcmap" ||
+              dspt == "absmem" ||
+              dspt == "additional_mapping_control_assume" ||
+              dspt == "func_arg" ||
+              dspt == "func_result" ||
+              dspt == "post_value_holder"
+                )
+              wn_amc_wrapper_item.insert(expr);
+            else if(
+              dspt == "issue_decode" ||
+              dspt == "issue_valid"  ||
+              dspt == "start_condition" ||
+              dspt == "variable_map_assume"
+            )
+              wn_aspt_item.insert(expr);
+            else
+              ILA_ASSERT(false) << "Unknown dspt : " << dspt;
+          } // for expr
+        } // for assumptions
+
+      } // gen wire name
+
+      std::vector<std::pair<std::string,bool>> amc_design_item; // no wn here : in the submodule 
+      std::vector<std::pair<std::string,bool>> amc_wrapper_item; // in the top module
+      std::vector<std::pair<std::string,bool>> asst_item; // assertions
+      std::vector<std::pair<std::string,bool>> aspt_item; // assumptions
+      { // find number name --- smt name
+        // algo : go line by line, check in "wn_" or not, if so, add ow ignored
+        enum {BEFORE, SUBMODULE, TOPMODULE} state = BEFORE;
+        for (std::string line;  std::getline(smt_info.full_smt, line);) {
+          if ( state == BEFORE and
+              line.find("; yosys-smt2-module ") == 0) { // starts with it
+            // new module
+            auto modname = line.substr(strlen("; yosys-smt2-module "));
+            if (modname == vlg_top_module_name) 
+              state = SUBMODULE; // go to next state
+          } // find vlg module
+          else if (state == SUBMODULE) {
+            if ( extractSigDefFromLine(
+                  vlg_top_module_name, line, wn_amc_design_item,
+                  amc_design_item) ) { /* do nothing */ }
+            else if(line.find("; yosys-smt2-module ") == 0) {
+              auto modname = line.substr(strlen("; yosys-smt2-module "));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/app/Bo-Yuan-Huang/ILAng/pullRequest?prid=4555809)